clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -298,35 +298,55 @@ protected void removeFromStorageEngine(int partition, byte[] keyBytes, Delete de
     }
   }
 
-  /** @return what kind of storage operation to execute, if any. */
-  private StorageOperationType getStorageOperationType(int partition, ByteBuffer valuePayload, ByteBuffer rmdPayload) {
-    PartitionConsumptionState pcs = partitionConsumptionStateMap.get(partition);
+  /**
+   * This method determines which type of storage operation it is based on input value and RMD payload.
+   */
+  StorageOperationType getStorageOperationType(int partition, ByteBuffer valuePayload, ByteBuffer rmdPayload) {
```

Comment:
I feel the logic here is very confusing since it is handling both `PUT` and `DELETE` ops.
Is it possible to break down this function into two instead? Or passing the op type directly to indicate whether this op is `PUT` or `DELETE`, so that this function can clearly create two branches to avoid the confusing logic.

Reply:
Sure, I understand it will be very confusing for other people. I can actually split the logic.